test/Libraries/Microsoft.Extensions.AI.Abstractions.Tests/Functions/DelegatingAIFunctionTests.cs

```diff
@@ -32,6 +32,7 @@ public void DefaultOverrides_DelegateToInnerFunction()
         Assert.Same(expected.UnderlyingMethod, actual.UnderlyingMethod);
         Assert.Same(expected.AdditionalProperties, actual.AdditionalProperties);
         Assert.Equal(expected.ToString(), actual.ToString());
+        Assert.Same(expected, actual.GetService<ApprovalRequiredAIFunction>());
```

Comment:
Another observation, what should be the expected behavior when the current function is a `ApprovalRequiredAIFunction` wrapping another `ApprovalRequiredFunction`, for those scenarios we might want to have a dedicated `Delegating GetService` behavior, that prioritizes Inner matches instead of itself?

Reply:
> what should be the expected behavior when the current function is a ApprovalRequiredAIFunction wrapping another ApprovalRequiredFunction

Once GetService exists, when would that realistically happen? My assumption is when it does after that, the dev actually wants their wrapper recognized and not skipped, though in practice even if they didn't, I'm not sure what it would impact.